Source/WebCore/rendering/svg/RenderSVGImage.cpp

```diff
@@ -344,13 +344,26 @@ void RenderSVGImage::imageChanged(WrappedImagePtr newImage, const IntRect* rect)
     if (renderTreeBeingDestroyed())
         return;
 
-    // The image resource defaults to nullImage until the resource arrives.
-    // This empty image may be cached by SVG resources which must be invalidated.
-    if (auto* resources = SVGResourcesCache::cachedResourcesForRenderer(*this))
-        resources->removeClientFromCache(*this);
+    bool invalidateLegacyResources = false;
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    invalidateLegacyResources = !document().settings().layerBasedSVGEngineEnabled();
+#endif
+
+    if (invalidateLegacyResources) {
+        // The image resource defaults to nullImage until the resource arrives.
+        // This empty image may be cached by SVG resources which must be invalidated.
+        if (auto* resources = SVGResourcesCache::cachedResourcesForRenderer(*this))
+            resources->removeClientFromCache(*this);
 
-    // Eventually notify parent resources, that we've changed.
-    LegacyRenderSVGResource::markForLayoutAndParentResourceInvalidation(*this, false);
+        // Eventually notify parent resources, that we've changed.
+        LegacyRenderSVGResource::markForLayoutAndParentResourceInvalidation(*this, false);
+    }
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
```

Comment:
The if/endig is not absolutely needed, it could just be an else section.

Reply:
`repaintClientsOfReferencedSVGResources` only exists if `LAYER_BASED_SVG_ENGINE` is defined.

Reply:
So I'd still need an if/endif... or did I misunderstand?

Reply:
Ah, did not know about repaintClientsOfReferencedSVGResources. Still, it seems now we behave differently when LAYER_BASED_SVG_ENGINE is not defined?

Reply:
True, I wanted to give it a try without forcing layouts. Can't really evaluate the impact at this point, we need more 'unskipped' tests first...